docs/Style-Guide.md

```diff
@@ -68,6 +69,93 @@ When developing any new feature, make sure to add or update the respective tests
+
+#### Type 1 - while filling a form
+
+These messages (mostly errors) are displayed whenever the user makes a mistake when filling out a form input. Examples:
+
+> A {fieldname} is required. —— Ex: A password is required.  
+> A {fieldname} cannot {action}. —— Ex: A username cannot contain spaces.  
+
+Because we do inline validations, the error message is shown whenever the input is invalid. To avoid showing the error too earlier, this is, when the user is still typing their answer, we delay the error validation a bit, to make sure the user has finished typing.
+
+```pug
+label.field
+  i18n.label Email
+  input.input(
+    :class='{error: $v.form.email.$error}'
+    name='email'
+    type='email'
+    v-model='form.email'
+    @input='debounceField("email")'
+    @blur='updateField("email")'
+    v-error:email='{ attrs: { "data-test": "badEmail" } }'
+  )
+```
+
+```js
+import { validationMixin } from 'vuelidate'
+import validationsDebouncedMixins from '@view-utils/validationsDebouncedMixins.js'
+
+mixins: [
+  validationMixin,
+  validationsDebouncedMixins // include debounceField and updateField
+]
+validations: {
+  form: {
+    email: {
+      [L('An email is required.')]: required,
+      [L('Please enter a valid email.')]: email,
+    }
+  }
+}
+```
+
+- Use `v-error` to automatically show the error message on the correct place. Read `vError.js` for more examples.
+- With `debounceField`, the error message (ex: _'Please enter a valid email.'_) is only shown when the user stopped typing. `updateField` shows the error when the user changed the focus no another input (on blur).
+
+Using the "debounce approach" isn't needed in all inputs. Usually it's recommended when a message error doesn't make sense while the user is still typing their answer, such as an e-mail.
+
+#### Type 2 - User action
+
+These banners let the user know that a certain action was completed successfully OR that an error occurred, and what they can do about it. These errors are more general than form errors, because they concern the action as a whole.
+If there was an error, we should tell what's wrong whenever possible, what action should be done and ask them to try again.
+
+```pug
+banner-scoped(ref='formMsg')
+```
+
+```js
+import BannerScoped from '@components/BannerScoped.vue'
+
+submit () {
+  // If everything went right...
+  this.$refs.formMsg.success(L('Changes saved!'))
+
+  // If something wen't wrong... Try to be more specific whenever possible.
+  this.$refs.formMsg.danger(L('Something went wrong, please try again.'))
+}
+```
+
+#### Type 3 - App Error
+
+This banner informs the user that there's something going on with the app itself, or with something that is not related with a user action. Example:
+
+```html
+<!-- index.html -->
+<banner-general ref="bannerGeneral"></banner-general>
+```
+
+```js
+// main.js
+import BannerGeneral from './views/components/BannerGeneral.vue'
+
+this.$refs.bannerGeneral.show(L('Trying to reconnect...'), 'wifi')
+```
+
```

Comment:
Excellent work! This is very helpful now. :-)

I would make only two more suggestions for now:

1. Add an empty newline after line 73 for consistent headings formatting
2. Add a "NOTE: this section is under construction and will change soon" message somewhere here in the "Type 3" section, since we'll need to add a (most likely an `sbp`-based) API to update the global banner